source/module_base/atom_in.h

```diff
@@ -91,7 +91,23 @@ class atom_in
             {"Meitnerium", 109},  {"Darmstadtium", 110},  {"Roentgenium", 111}, {"Copernicium", 112}, {"Nihonium", 113},    {"Flerovium", 114},
             {"Moscovium", 115},   {"Livermorium", 116},   {"Tennessine", 117},  {"Oganesson", 118}
           };             
-          
+    std::map<std::string, int> principle_quantum_number
```

Comment:
the principle quantum number is needed for hydrogen-like orbital generation if strategy is full or minimal. Say for element Ta, if full, program gets its principle quantum number as 6 and will generate 1s2s3s4s5s6s, 2p3p4p5p6p, 3d4d5d6d, 4f5f6f, 5g6g, 6h orbitals, if minimal, then 1s2p3d4f5g6h. The reason why minimal is designed is because the number of nodes. Orbitals generated with minimal strategy will be nodeless, this will guarantee a relatively larger overlap rather than the smaller which may partially result from the cancellation between different parts of orbital, because our NAO always has fewer nodes

Reply:
I will expose a parameter `qo_strategy` to user to tune the strategy of generating hydrogen-like orbitals